homeworks/06-advanced-redux/src/middleware/logger.ts

```diff
@@ -0,0 +1,48 @@
+import { BasketActions } from '../redux/basket/basket.types';
+import { ItemsActions } from '../redux/items/items.types';
+import { Action, Middleware } from 'redux';
+
+type LoggedActions = BasketActions.AddToBasket | BasketActions.RemoveFromBasket | ItemsActions.PizzaViewed;
+
+type event = {
+    eventName: LoggedActions;
+    pizzaName?: string;
+    pizzaPrice?: number;
+};
+export interface Dispatch {
+    <A extends Action>(action: A): A;
+}
+
+export const logger: Middleware = () => next => action => {
+    switch (action.type) {
```

Comment:
do you really need switch here?

Reply:
Well, i could've put actions that i need to log into an array and then check if it includes the passed action type. I guess my main argument for the switch here is that it is more in the redux way of thinking. 
If i would have more actions to log (like 5 or more) i would definitely move them into an array.